crates/bevy_render/src/gpu_component_array_buffer.rs

```diff
@@ -20,7 +20,7 @@ impl<C: Component + GpuArrayBufferable> Plugin for GpuComponentArrayBufferPlugin
         if let Ok(render_app) = app.get_sub_app_mut(RenderApp) {
             render_app.add_systems(
                 Render,
-                prepare_gpu_component_array_buffers::<C>.in_set(RenderSet::Prepare),
+                prepare_gpu_component_array_buffers::<C>.in_set(RenderSet::PrepareAssets),
```

Comment:
This is not what is intended. Per-frame dynamic data is meant to still be prepared in prepare.

Reply:
You're right that this shouldn't be in PrepareAssets (doesn't make any sense) but the GpuArrayBufferIndex is currently consumed in queue_material_meshes so either we need to break that dependency so we can place this system in prepare or find another place to put it.

Reply:
The reason that it is used in queue_material_meshes is so that it can be used in the sort key.

Reply:
Right, this comes to the key point of the PR. I had implemented this before as queuing with a default dynamic offset value (I think maybe None) and then filling it in in prepare, where batching will be done for material meshes. Now that we have `GpuArrayBuffer` and the `GpuComponentArrayBufferPlugin` is being used for `MeshUniform`, this code path changed. I think we will have to stop using `GpuComponentArrayBufferPlugin` for `MeshUniform` and instead make a custom prepare_mesh_uniforms system that does what the plugin does, but also updates the dynamic offset of the corresponding phase item.

Reply:
Implemented outside the plugin in `prepare_mesh_uniform_buffer`